soh/soh/Enhancements/debugger/actorViewer.cpp

```diff
@@ -0,0 +1,1261 @@
+#include "actorViewer.h"
+#include "../../util.h"
+#include "../libultraship/SohImGuiImpl.h"
+#include "ImGuiHelpers.h"
+
+#include <array>
+#include <bit>
+#include <map>
+#include <string>
+
+extern "C" {
+#include <z64.h>
+#include "z64math.h"
+#include "variables.h"
+#include "functions.h"
+#include "macros.h"
+extern GlobalContext* gGlobalCtx;
+
+#include "textures/icon_item_static/icon_item_static.h"
+#include "textures/icon_item_24_static/icon_item_24_static.h"
+}
+
+typedef struct {
+    u16 id;
+    u16 params;
+    Vec3f pos;
+    Vec3s rot;
+} ActorInfo;
+
+typedef enum {
+    LIST,
+    TARGET,
+    HELD,
+    INTERACT
+} RetrievalMethod;
+
+std::vector<std::string> acMapping = { 
```

Comment:
Since you only use the cstring out of this and don't add or remove from this it should be
`std::array<const char*, iDontFeelLikeCountingThose>`